lib/bolt/transport/sudoable/connection.rb

```diff
@@ -70,6 +70,15 @@ def execute(*_args)
           message = "#{self.class.name} must implement #{method} to execute commands"
           raise NotImplementedError, message
         end
+
+        def build_command(sudo_str, command_str, sudo_id, input_stdin)
```

Comment:
This is a fairly generic name to be on a mixin. It's clear in the context of this module, but once it's included in another class, it's not so obvious what its function is. It might be helpful to include the word `sudo` in there somewhere.

It might also make sense to split out the part that generates `sh -c '...'` from the part that puts `sudo_str` in front of it. Since the latter is just string concatenation, maybe the only thing this function needs to do is generate the `sh` invocation and let the caller handle prepending the run-as command.

Reply:
I added more descriptive names and separated out appending the command to echo the sudo_id before the command. I left both methods in the Sudoable::Connection class because the same logic is used for SSH::connection and Local::Shell.